File: t1/SEP3BlazorT1Client/SEP3BlazorT1Client/Models/Host.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using Newtonsoft.Json;

namespace SEP3BlazorT1Client.Models
{
    public class Host : User
    {
        [JsonProperty("hostReviews", NullValueHandling = NullValueHandling.Ignore)]
        public IList<HostReview>? HostReviews { get; set; } = new List<HostReview>();
        [JsonProperty("profileImageUrl")]
        public string ProfileImageUrl { get; set; }
        [Required(ErrorMessage = "Please enter your CPR number")]
        [JsonProperty("cpr")]
        public string Cpr { get; set; }
        [JsonProperty("isApprovedHost")]
        public bool IsApprovedHost { get; set; } = false;

        public override string ToString()
        {
            return
                $"{Id} {FirstName} {LastName} {PhoneNumber} {Email} {Password} {Cpr} {ProfileImageUrl} {IsApprovedHost}";
        }
    }
}

File: t3/viabnb.tier3.dataserver.rest/src/main/java/dk/viabnb/sep3/group6/dataserver/rest/t3/models/Host.java
package dk.viabnb.sep3.group6.dataserver.rest.t3.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.List;

public class Host extends User
{
  private List<HostReview> hostReviews;
  private String cpr;
  private boolean isApprovedHost;

  public Host(int id, String email, String password, String firstName,
      String lastName, String phoneNumber, String profileImageUrl,
      List<HostReview> hostReviews, String cpr, boolean isApprovedHost)
  {
    super(id, email, password, firstName, lastName, phoneNumber,
        profileImageUrl);
    this.hostReviews = hostReviews;
    this.cpr = cpr;
    this.isApprovedHost = isApprovedHost;
  }

  public List<HostReview> getHostReviews()
  {
    return hostReviews;
  }

  public void setHostReviews(List<HostReview> hostReviews)
  {
    this.hostReviews = hostReviews;
  }

  public String getCpr()
  {
    return cpr;
  }

  public void setCpr(String cpr)
  {
    this.cpr = cpr;
  }

  @JsonProperty("isApprovedHost") public boolean isApprovedHost()
  {
    return isApprovedHost;
  }

  public void setApprovedHost(boolean approvedHost)
  {
    isApprovedHost = approvedHost;
  }
}
